lib/rules/no-misleading-character-class.js

```diff
@@ -73,80 +72,120 @@ function isUnicodeCodePointEscape(char) {
+            !isUnicodeCodePointEscape(previous) &&
+            !isUnicodeCodePointEscape(char)
+        ) {
+            return [previous, char];
+        }
+
+        return null;
     },
 
-    surrogatePair(chars) {
-        return chars.some((c, i) => {
-            if (i === 0) {
-                return false;
-            }
-            const c1 = chars[i - 1];
-
-            return (
-                isSurrogatePair(c1.value, c.value) &&
-                (
-                    isUnicodeCodePointEscape(c1) ||
-                    isUnicodeCodePointEscape(c)
-                )
-            );
-        });
+    surrogatePair(char, index, chars) {
+        if (index === 0) {
+            return null;
+        }
+
+        const previous = chars[index - 1];
+
+        if (
+            isSurrogatePair(previous.value, char.value) &&
+            (
+                isUnicodeCodePointEscape(previous) ||
+                isUnicodeCodePointEscape(char)
+            )
+        ) {
+            return [previous, char];
+        }
+
+        return null;
     },
 
-    combiningClass(chars) {
-        return chars.some((c, i) => (
-            i !== 0 &&
-            isCombiningCharacter(c.value) &&
-            !isCombiningCharacter(chars[i - 1].value)
-        ));
+    combiningClass(char, index, chars) {
+        if (
+            index !== 0 &&
+            isCombiningCharacter(char.value) &&
+            !isCombiningCharacter(chars[index - 1].value)
+        ) {
+            return [chars[index - 1], char];
+        }
+
+        return null;
     },
 
-    emojiModifier(chars) {
-        return chars.some((c, i) => (
-            i !== 0 &&
-            isEmojiModifier(c.value) &&
-            !isEmojiModifier(chars[i - 1].value)
-        ));
+    emojiModifier(char, index, chars) {
+        if (
+            index !== 0 &&
+            isEmojiModifier(char.value) &&
+            !isEmojiModifier(chars[index - 1].value)
+        ) {
+            return [chars[index - 1], char];
+        }
+
+        return null;
     },
 
-    regionalIndicatorSymbol(chars) {
-        return chars.some((c, i) => (
-            i !== 0 &&
-            isRegionalIndicatorSymbol(c.value) &&
-            isRegionalIndicatorSymbol(chars[i - 1].value)
-        ));
+    regionalIndicatorSymbol(char, index, chars) {
+        if (
+            index !== 0 &&
+            isRegionalIndicatorSymbol(char.value) &&
+            isRegionalIndicatorSymbol(chars[index - 1].value)
+        ) {
+            return [chars[index - 1], char];
+        }
+
+        return null;
     },
 
-    zwj(chars) {
-        const lastIndex = chars.length - 1;
+    zwj(char, index, chars) {
+        if (
+            index !== 0 &&
+            index !== chars.length - 1 &&
+            char.value === 0x200d &&
+            chars[index - 1].value !== 0x200d &&
+            chars[index + 1].value !== 0x200d
+        ) {
+            return chars.slice(index - 1, index + 1);
```

Comment:
```suggestion
            return chars.slice(index - 1, index + 2);
```

To return all three characters.

Reply:
Aha, thank you!